backend/src/bundles/lms-data/lms-data.service.ts

```diff
@@ -0,0 +1,119 @@
+import { Headers } from 'node-fetch';
+
+import { userRepository } from '~/bundles/users/users.js';
+import { ErrorMessage } from '~/common/enums/enums.js';
+import { HttpCode, HttpError } from '~/common/http/http.js';
+import { config } from '~/common/packages/config/config.js';
+import { http } from '~/common/packages/http/http.js';
+import { type Service } from '~/common/types/types.js';
+
+import { parseLMSServerData } from './helpers/helpers.js';
+import { LMSDataEntity } from './lms-data.entity.js';
+import { type LMSDataRepository } from './lms-data.repository.js';
+import {
+    type LMSDataServerResponseDto,
+    type UserLMSDataDto,
+} from './types/types.js';
+
+class LMSDataService implements Service {
+    private lmsDataRepository: LMSDataRepository;
+    private requestsToLMSHeaders: { 'X-Token': string };
```

Comment:
Provide just token from env to private field and put this token to request when you make request